### DISCUSSION TITLE
Schedule Transactions used for Secure / Safe Trade Standard

### DISCUSSION THREAD

**Discussion by AdrianKBL** [hiero-ledger/hiero-improvement-proposals#731] [Ideas]
**_Summary:_**

This proposal aims to standardize the way of generating the codes of a **Schedule Transaction**, so for example, when we use tools such as **Secure Trade** (HashPack) or **Safe Trade** (Kabila), they are compatible.

_**Motivation:**_

Right now if someone tries to send one or more **NFTs through Schedule Transactions**, either from Kabila to HashPack, from HashPack to Kabila or from HashPack to Blade or any other Wallet, it is not supported as we use different algorithms to obfuscate the generated code. . We believe that this is a great barrier for users, we have to make their lives easier and in this case we are making it more difficult for them.

_**Specification:**_

From Kabila, we believe that the correct way to standardize this, would be to use as the generated code:

`scheduleId-accReceiver-timeStampCreated and obfuscate it to base64.`

These parameters are necessary, so when the person who receives them introduces the code, through the accReceiver we can know if the code is correct for that AccountId, so he doesn't have to pay the fee to consult the scheduleId, the timeStampCreated, to know if it has expired (if it has expired we do not consult the network and we do not pay fees).

**_Backwards Compatibility_**

This proposal is expected to be backwards compatible, as it only introduces a new way to standardize how wallets or applications generate these codes.

**Comment by mshakeg** [comment]
@AdriMsM91 I wouldn't say the following is directed to the idea you're proposing, but more around the idea of scheduled transactions. Does it not open up the possibility for MEV on Hedera? As it publicises when the Hedera network will process a transaction well in advance.

**Comment by teacoat** [comment]
We are open to a standard at hashpack, though our big concern around this is fee display in other wallets. Fee's can be set up quite maliciously and without a comprehensive fee breakdown or a fee-type blacklist of this could be used to scam users.

**Comment by nickpoorman** [comment]
@AdriMsM91 - What's the use case here? I'm not clear on what the goal is or why. Are you saying that you want to standardize putting `scheduleId-accReceiver-timeStampCreated` in a memo field? If so, what's the reason for that?

**Reply by AdrianKBL** [reply]
Hello @nickpoorman this is nothing related about the memo, this is related about how Wallets can share the different codes of Schedule Transactions, such as Kabila / HashPack / Blade...

![image](https://github.com/hashgraph/hedera-improvement-proposal/assets/59306428/17135d25-c68f-4a7e-b7d1-7cb01aacef2f)
![image](https://github.com/hashgraph/hedera-improvement-proposal/assets/59306428/b909a668-427f-4d11-93c6-a7ec870ee713)

As you can see the Code generated by HashPack wallet isn't valid for Kabila Wallet and viceverse. This is creating a really big GAP for new users, then you need to explain them that they need to import the Account in the other wallet If they want to transfer NFTs with FallBack Fees otherwise there isn't any way...etc.

We faced many many big problems with this, that's why we think it's really important.
